src/BotPlutusInterface/Config.hs

```diff
@@ -0,0 +1,281 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+{-# OPTIONS -fno-warn-orphans  #-}
+
+module BotPlutusInterface.Config (
+  docPABConfig,
+  loadPABConfig,
+  savePABConfig,
+) where
+
+import BotPlutusInterface.Effects (
+  ShellArgs (..),
+  callLocalCommand,
+ )
+import BotPlutusInterface.Types (CLILocation (..), LogLevel (..), PABConfig (..))
+
+import Cardano.Api (ExecutionUnits (..), NetworkId (Mainnet, Testnet), unNetworkMagic)
+import Config (Section (Section), Value (Atom, Sections, Text))
+import Config.Schema (
+  HasSpec (anySpec),
+  ValueSpec,
+  atomSpec,
+  generateDocs,
+  naturalSpec,
+  sectionsSpec,
+  trueOrFalseSpec,
+  (<!>),
+ )
+import Data.Default (def)
+import Data.String.ToString (toString)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import PlutusConfig.Base (
+  enumToAtom,
+  filepathSpec,
+  maybeSpec,
+  pathSpec,
+  portSpec,
+ )
+import PlutusConfig.Cardano.Api ()
+import PlutusConfig.Cardano.Api.Shelley (
+  readProtocolParametersJSON,
+ )
+import PlutusConfig.Ledger ()
+import PlutusConfig.Types (
+  ToValue (toValue),
+  deserialize,
+  sectionWithDefault,
+  sectionWithDefault',
+  serialize,
+  withNamePrefixSpec,
+ )
+import Prelude
+
+instance ToValue CLILocation where
+  toValue Local = Atom () "local"
+  toValue (Remote url) = Text () url
+
+cliLocationSpec :: ValueSpec CLILocation
+cliLocationSpec =
+  Local <$ atomSpec "local"
+    <!> Remote <$> withNamePrefixSpec "destination" anySpec
+
+instance ToValue LogLevel where
+  toValue = enumToAtom
+
+logLevelSpec :: ValueSpec LogLevel
+logLevelSpec =
+  Error <$ atomSpec "error"
+    <!> Warn <$ atomSpec "warn"
+    <!> Notice <$ atomSpec "notice"
+    <!> Info <$ atomSpec "info"
+    <!> Debug <$ atomSpec "debug"
+
+instance ToValue (Integer, Integer) where
+  toValue = toValue . forceBudgetToExecutionUnits
+
+instance HasSpec (Maybe (Integer, Integer)) where
+  anySpec = maybeSpec (executionUnitsToForceBudget <$> anySpec)
+
+forceBudgetToExecutionUnits :: (Integer, Integer) -> ExecutionUnits
```

Comment:
Doesn't look it needed anymore as force budget option removed. Couldn't find it used anywhere also.